docs/sources/style-guide/write-for-developers/index.md

```diff
@@ -0,0 +1,148 @@
+
+### Elements of API references
+
+Properly document the most common elements of an API reference, such as the title, parameters, return values, and so on. The following suggestions will help you to write more complete and consistent documentation:
+
+| Element                                               | Description                                                           |
+| ------------------------------------------------- | ----------------------------------------------------------------------- |
+| Title and description       | The name of the element and a description of it in one or two sentences. Put API names, classes, methods, etc. in `code font`. |
+| Syntax | The code signature that defines the element. If multiple programming languages can be used, give the syntax in each. Put the signature in `code font`. |
+| Parameters | If the element has parameters, specify their descriptions, data types, and state whether they are optional or required. Put the parameters in _italic_. |
+| Return values| If the element returns a value, describe the range of possibilities and the data type. |
+| Error codes| Describe errors or exceptions and the conditions under which they occur. |
+| Comments | Describe any important information that hasn't been previously included in the title, description, syntax, parameters, or return values. For example, you may explain non-obvious context, compare it to similar elements, or provide cautionary notes about potential gotchas. |
+
+Additional tips:
+- Remember to write concisely. Don't say "This method does this..." or "This parameter describes that..."
+- When the names of code elements are singular, don't make them plural. Instead, add a plural noun to describe them. For example, don't change `MyEvent` to `MyEvents`; refer to the `MyEvent` objects. 
+- If the element does some sort of action, start the first sentence of the description with an action verb. 
+
+## Code examples
+
+It is common for readers of documentation to skim through it to find code samples they can copy and paste and run "as is". For this reason, it is worth taking the time to ensure that whenever possible the examples you provide are production-ready.
+
+Of course, this doesn't mean you should avoid writing code examples intended merely to illustrate a point. When your intent is to explain something, it isn't necessary for your code to be runnable so long as the developer can learn from it to do something similar on their own.
+
+When providing an example, give a written description. You can put it either in the body of the document or as explanatory comments within the example code.
+
+Remember the basic rule: your explanation of the code shouldn't describe _what_ it does, but _why_ it does what it does. (Refer to [_Docs for Developers_](https://docsfordevelopers.com/)).
```

Comment:
Why are you linking to a book that must to be purchased?  If the information is not freely available on the web, I would try to find another, more accessible reference to link to.  AND link to the specific topic you have in mind.  Pointing someone to a 250 page book isn't very helpful.

Reply:
I don't think linking to a book -- a recent one that is the only significant book to focus directly on the topic of technical writing for developer documentation -- is inappropriate here. No one needs to buy a book if they don't want to. The point is the advice isn't being pulled out of a hat, but somewhere credible that someone can research further. Still, I can look for alternative sources on the web (but then you could easily run into the credibility problem. Why is that opinion worth following?)

Reply:
So, I found a blog post that says basically the same thing at https://blog.codinghorror.com/code-tells-you-how-comments-tell-you-why/

The author is a developer who lives with hit wife, two cats, lots of computers, and is currently studyig Ruby. 

I would rather support this claim by reference to the leading book in the field. But if someone finds a better cite, let's use it.

Reply:
I don't think the statement requires a reference. It's very general, common sense and as long as it isn't a quote, I don't think you need to cite anything. I'd add any helpful reference materials in a Learn more type section if it were me.